include/librealsense2/h/rs_types.h

```diff
@@ -114,11 +114,24 @@ typedef enum rs2_extension
     RS2_EXTENSION_POSE_FRAME,
     RS2_EXTENSION_POSE_PROFILE,
     RS2_EXTENSION_TM2,
+    RS2_EXTENSION_SOFTWARE_DEVICE,
+    RS2_EXTENSION_SOFTWARE_SENSOR,
     RS2_EXTENSION_COUNT
 } rs2_extension;
 const char* rs2_extension_type_to_string(rs2_extension type);
 const char* rs2_extension_to_string(rs2_extension type);
 
+/** \brief Specifies types of different matchers */
+typedef enum rs2_matchers
+{
+   RS2_MATCHER_DI, //depth, ir
+   RS2_MATCHER_DI_C, //depth, ir, color
+   RS2_MATCHER_DLR_C, //depth, left ir, right ir, color
+   RS2_MATCHER_DLR, //depth, left ir, right ir
+   RS2_MATCHER_DEFAULT, //default matcher
+   RS2_MATCHER_COUNT
+}rs2_matchers;
+
```

Comment:
since the user is expected to use these types - can you please elaborate in the comment how each matcher compares the images (based on exact TS / frame number / closest TS...) so the user can select what's best for him? I would also add a hint in the comment for device type and the relevant matcher, for common usages.